### PR TITLE
Article Fetching

### PR DESCRIPTION
**Pull request by btburton42**
Delivers working backend server fetching for articles and next article. Tears out lots of outdated code. Cleans up components.

### CODE REVIEW

File: app/actions/article.js [hunk 1]
```diff
@@ -1,23 +1,27 @@
...
-export const ACTIVATE_TOPIC = 'ACTIVATE_TOPIC';
-export const NEW_QUESTIONS = 'NEW_QUESTIONS';
-export const NEW_ARTICLE = 'NEW_ARTICLE';
+export function fetchArticle(articleId) {
+  return (dispatch) => {
+    dispatch({ type: 'FETCH_ARTICLE', articleId});
 
-export function addHighlight(start, end, selectedText) {
-  return { type: ADD_HIGHLIGHT, selection: {start, end, selectedText} };
+    return fetch(`http://localhost:5000/api/articles/${articleId}/?format=json`) // TODO: resolve this absolute URL issue with backend
```
Comment (stefanv) [inline app/actions/article.js:5]: What do the backticks mean in JavaScript--I've never run across that before!

**Comment by stefanv** [comment]
I gave this a quick read, and overall looks good to me.

File: app/actions/article.js (hunk 1 above)
Comment (btburton42) [inline app/actions/article.js:5]: That's ES6 string interpolation. So I can pass in `${articleId}` as a var.

Comment (stefanv) [inline app/actions/article.js:5]: That's very useful to know--thanks!

**Pull request merged**